src/random_read_worker.rs

```diff
@@ -67,24 +67,39 @@ pub struct RandomReadWorker {
     endpoint: Url,
     client: Client,
     query_gen: Box<dyn QueryGenerator + Send>,
+    behav: WorkerBehaviour,
 }
 
 impl RandomReadWorker {
-    pub fn new(query_gen: Box<dyn QueryGenerator + Send>, endpoint: Url) -> Self {
+    pub fn new(query_gen: Box<dyn QueryGenerator + Send>, endpoint: Url, behav: WorkerBehaviour) -> Self {
         let client = Client::builder().tcp_nodelay(true).build().unwrap();
 
-        Self { endpoint, client, query_gen }
+        Self { endpoint, client, query_gen, behav }
     }
 
-    async fn measure_query(query: impl Future<Output = reqwest::Result<Response>>) -> reqwest::Result<Duration> {
+    async fn measure_query(
+        behav: WorkerBehaviour,
+        query: impl Future<Output = reqwest::Result<Response>>,
+    ) -> reqwest::Result<Option<Duration>> {
         let start = Instant::now();
 
-        let resp = query.await?.error_for_status()?.text().await?;
-        std::hint::black_box(resp);
-
-        let end = Instant::now();
-
-        Ok(end.duration_since(start))
+        match query.await {
+            Ok(resp) => {
+                let resp = resp.error_for_status()?;
+
+                match resp.text().await {
+                    Ok(text) => {
+                        std::hint::black_box(text);
+                        let end = Instant::now();
+                        Ok(Some(end.duration_since(start)))
+                    },
+                    Err(_) if behav == WorkerBehaviour::IgnoreConnectionError => Ok(None),
+                    Err(e) => Err(e),
+                }
+            },
+            Err(_) if behav == WorkerBehaviour::IgnoreConnectionError => Ok(None),
+            Err(e) => Err(e),
+        }
```

Comment:
They continue sending requests. I did this to avoid potential synchronization issues that lead to errors being falsely reported.

For example, if there was a notification scheme between the workers (that tells them if the server is running).

1. kill worker kills server
2. kill worker notifies others
3. read worker sends query (because it didn't get the notification yet)
4. read worker assumes server is broken, you get an error printed (this error is the false-positive error)
5. read worker sees that the server is down

Reply:
Sorry. One two more questions about this. 
First, in the new verification mode, we do not care about QPS, right? 
Second, does a worker send the same query while the server is down (similar to updates)?

Reply:
1. technically no, but nice to see how the server performs under heavy read-write load
2. yes, the readers continue to send queries but the qps measurements are discarded for failed queries (in verify mode with periodic kill, in normal verify mode they are not allowed to fail)